blockchain/blockchain.go

```diff
@@ -703,3 +967,20 @@ func (b *Blockchain) StoreGenesis(
 func (b *Blockchain) WriteRunningEventFilter() error {
 	return b.runningFilter.Write()
 }
+
+func (b *Blockchain) Stop() error {
+	if b.trieDB.Scheme() == triedb.PathScheme && b.StateFactory.UseNewState {
+		head, err := b.HeadsHeader()
+		if err != nil {
+			return err
+		}
+
+		stateUpdate, err := b.StateUpdateByNumber(head.Number)
+		if err != nil {
+			return err
+		}
+
+		return b.trieDB.Journal(stateUpdate.NewRoot)
```

Comment:
Question: What happens if synchronizer writes a new block in the middle of this process?